applying-stpa/xlsx-yaml.py

```diff
@@ -0,0 +1,134 @@
+import yaml
+import pandas as pd
+
+def convert_UCAs_to_yml(xlsx_file, uca_sheet):
+    uca_df = pd.read_excel(xlsx_file, engine='openpyxl', sheet_name=uca_sheet)
+    scenario_df = pd.read_excel(xlsx_file, engine='openpyxl', sheet_name='Level 1 Loss Scenario Analysis')
+    uca_scenarios = {}
+    curr_key = ""
+    for row in scenario_df.iterrows():
+        data = row[1]
+        if type(data["UCA"]) is str:
+            curr_key = data["UCA"].split(": ")
+            curr_key = curr_key[0].strip()
+            uca_scenarios.update({curr_key: {"ls1":[], "ls2":[]}})
+        if type(data["Loss Scenario Type 1"]) is str:
+            uca_scenarios[curr_key]['ls1'].append(data["Loss Scenario Type 1"])    
+        if type(data["Loss Scenario Type 2"]) is str:
+            uca_scenarios[curr_key]["ls2"].append(data["Loss Scenario Type 2"])
+    uca_categories = ["Providing", "Not Providing", "Timing", "Duration"]
+    cc_labels = {"Providing":"P", "Not Providing":"NP", "Timing": "T", "Duration": "D"}
+    components = []
+    component = {}
+    control_action = {}
+    for row in uca_df.iterrows():
+        data = row[1]
+        if type(data['Controller']) is str:
+            if component:
+                component["Control Actions"].append(control_action)
+                components.append(component)
+            component = {
+                "Text": data['Controller'],
+                "Control Actions": [],
+                "Controller Constraints": []
+            }
+            control_action = {}
+        if type(data['Control Action']) is str:
+            if control_action:
+                component["Control Actions"].append(control_action)
+            split_control_action = data['Control Action'].split(": ")
+            control_action = {
+                "Identifier": split_control_action[0].strip(),
+                "Text": split_control_action[1].strip(),
+                "Unsafe Control Actions": []
+            }
+        for category in uca_categories:
+            if type(data[category]) is str:
+                if not data[category].strip():
+                    continue
+                split_uca = data[category].split(": ")
+                control_action["Unsafe Control Actions"].append({
+                    "Identifier": split_uca[0].strip(),
+                    "Text": split_uca[1].strip(),
+                    "Reason": category
+                })
+                if split_uca[0].strip() in uca_scenarios.keys():
+                    control_action["Unsafe Control Actions"][-1].update({
+                        "Scenarios": []
+                    })
+                    for ls1 in uca_scenarios[split_uca[0].strip()]['ls1']:
+                        identifier = "LS-1-"+split_uca[0].strip()
```

Comment:
The way these identifiers are created needs to be updated, (perhaps using a `Loss Scenario identifier` variable with a counter)